state-sync/aptos-data-client/src/aptosnet/mod.rs

```diff
@@ -503,95 +537,163 @@ impl DataSummaryPoller {
-                    update_advertised_data_metrics(global_data_summary);
+    /// Attempts to fetch the next regular peer to poll from the data client.
+    /// If an error is encountered, the error is logged and None is returned.
+    fn try_fetch_regular_peer(&self) -> Option<PeerNetworkId> {
+        self.data_client
+            .fetch_regular_peer_to_poll()
+            .unwrap_or_else(|error| {
+                log_poller_error(error);
+                None
+            })
+    }
+}
+
+/// Logs the given poller error based on the logging frequency
+fn log_poller_error(error: Error) {
+    sample!(
+        SampleRate::Duration(Duration::from_secs(POLLER_LOG_FREQ_SECS)),
+        error!(
+            (LogSchema::new(LogEntry::StorageSummaryRequest)
+                .event(LogEvent::PeerPollingError)
+                .message("Unable to fetch peers to poll!")
+                .error(&error))
+        );
+    );
+}
+
+/// Updates the metrics for the number of in-flight polls
+fn update_in_flight_metrics(label: &str, num_in_flight_polls: u64) {
+    sample!(
+        SampleRate::Frequency(IN_FLIGHT_METRICS_SAMPLE_FREQ),
+        set_gauge(
+            &metrics::IN_FLIGHT_POLLS,
+            label.into(),
+            num_in_flight_polls,
+        );
+    );
+}
+
+/// Spawns a dedicated poller for the given peer.
+fn poll_peer(data_client: AptosNetDataClient, peer: PeerNetworkId, runtime: Option<Handle>) {
+    // Create the poller for the peer
+    let poller = async move {
+        // Start the peer polling timer
+        let timer = start_timer(
+            &metrics::REQUEST_LATENCIES,
+            StorageServiceRequest::GetStorageServerSummary
+                .get_label()
+                .into(),
+        );
+
+        // Fetch the storage summary for the peer and stop the timer
+        let result: Result<StorageServerSummary> = data_client
+            .send_request_to_peer_and_decode(peer, StorageServiceRequest::GetStorageServerSummary)
+            .await
+            .map(Response::into_payload);
+        drop(timer);
+
+        // Check the storage summary response
+        let storage_summary = match result {
+            Ok(storage_summary) => storage_summary,
+            Err(error) => {
+                error!(
+                    (LogSchema::new(LogEntry::StorageSummaryResponse)
+                        .event(LogEvent::PeerPollingError)
+                        .message("Error encountered when polling peer!")
+                        .error(&error)
+                        .peer(&peer))
                 );
+                return;
             }
-        }
-    }
+        };
+
+        // Update the global storage summary and the summary for the peer
+        data_client.update_summary(peer, storage_summary);
+        data_client.update_global_summary_cache();
+
+        // Mark the in-flight poll as now complete
+        data_client.in_flight_request_complete(&peer);
```

Comment:
Where are we marking the request as inflight?

Reply:
See above 😄